packages/pluggableWidgets/rating-native/src/lib/StarRating.tsx

```diff
@@ -0,0 +1,153 @@
+    fullStar: "star",
+    fullStarColor: "black",
+    halfStar: "star-half-o",
+    halfStarColor: undefined,
+    halfStarEnabled: false,
+    iconSet: "FontAwesome",
+    maxStars: DEFAULT_MAX_RATING,
+    rating: DEFAULT_RATING,
+    reversed: false,
+    starSize: 40,
+    starStyle: {},
+    selectedStar: undefined
+};
+
+interface Props extends StarRatingProps {
+    iconSet: keyof typeof iconSets;
+}
+
+class StarRating extends Component<Props> {
+    static defaultProps: typeof defaultProps;
+    starRef: Array<ClassicComponent<AnimatableProperties<ViewStyle> & ViewProps, any> | null>;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.starRef = [];
+        this.onStarButtonPress = this.onStarButtonPress.bind(this);
+    }
+
+    onStarButtonPress(rating: number) {
+        this.props.selectedStar?.(rating);
+    }
+
+    render() {
+        const {
+            activeOpacity,
+            animation,
+            buttonStyle,
+            containerStyle,
+            disabled,
+            emptyStar,
+            emptyStarColor,
+            fullStar,
+            fullStarColor,
+            halfStar,
+            halfStarColor,
+            halfStarEnabled,
+            iconSet,
+            maxStars,
+            rating,
+            reversed,
+            starSize,
+            starStyle
+        } = this.props;
+
+        const newContainerStyle = {
+            flexDirection: reversed ? "row-reverse" : "row",
+            justifyContent: "space-between",
+            ...StyleSheet.flatten(containerStyle)
+        } as ViewStyle;
+
+        let starsLeft = Math.round((rating ?? DEFAULT_RATING) * 2) / 2;
+        const starButtons = [];
+
+        for (let i = 0; i < (maxStars ?? DEFAULT_MAX_RATING); i++) {
+            let starIconName = emptyStar;
+            let finalStarColor = emptyStarColor;
+
+            if (starsLeft >= 1) {
+                starIconName = fullStar;
+                finalStarColor = fullStarColor;
+            } else if (starsLeft === 0.5) {
+                starIconName = halfStar;
+                finalStarColor = halfStarColor || fullStarColor;
+            }
+
+            const starButtonElement = (
+                <AnimatableView key={i} ref={node => (this.starRef[i] = node)}>
+                    <StarButton
+                        activeOpacity={activeOpacity}
+                        buttonStyle={buttonStyle}
+                        disabled={disabled}
+                        halfStarEnabled={halfStarEnabled}
+                        iconSet={iconSet}
+                        onStarButtonPress={(rating: number) => {
+                            if (animation && ANIMATION_TYPES.includes(animation)) {
+                                for (let s = 0; s <= i; s++) {
+                                    const component = this.starRef[s];
+                                    const millisecondsInSecond = 1000;
+                                    const offset = 200;
```

Comment:
While technically true, I'm still not sure what these 1000 ms mean. Judging by the code below, I'm guessing that 1000 ms is the base delay and 200 ms the additional delay per star (seen from the left) to produce some sort of "wave" animation? I was hoping for names that would explain this to the reader :). (And if they're truly constants, I would define them in uppercase in the outer scope.)